packages/devtools/lib/src/memory/memory.dart

```diff
@@ -285,9 +293,214 @@ class MemoryScreen extends Screen with SetStateMixin {
+                  text: 'Display Snapshot Classes from Library',
+                  c: 'collapsible-885 flex-no-wrap settings-left',
+                )..add([
+                    librariesUi = div(c: 'settings-libraries'),
+                  ]),
+                div(c: 'setttings-options')
+                  ..add([
+                    div(c: 'filter-pattern-area')
+                      ..add(<CoreElement>[
+                        span(
+                          text: 'Class Name Pattern: ',
+                          c: 'settings-class-pattern',
+                        ),
+                        classNameFilter = TextField(c: 'filter-class')
+                          ..value = memoryController.settings.pattern
+                          ..changed(_classPatternChanged),
+                        br(),
+                      ]),
+                  ]
+                    ..addAll(privateClasses = createCheckBox(
+                      'Hide Private Classes ',
+                      false,
+                      _liveUpdateFilters,
+                    ))
+                    ..addAll([
+                      br(), // Filter Option 3 available
+                      br(), // Filter Option 4 available
+                      br(), // Filter Option 5 available
+                    ])),
+                div(c: 'setttings-options-2')
+                  ..add([
+                    createCheckBox('Navigation Experiment '),
+                    br(), // Settings Option 2 available
+                    br(), // Settings Option 3 available
+                    br(), // Settings Option 4 available
+                    br(), // Settings Option 5 available
+                    br(), // Settings Option 6 available
+                  ]),
+                div()..flex(),
+                div(
+                  c: 'settings-buttons btn-group collapsible-785 '
+                      'nowrap margin-left text-right',
+                )
+                  ..flex()
+                  ..add(<CoreElement>[
+                    PButton('Apply')..click(_applySettings),
+                    PButton('Cancel')..click(_cancelSettings),
+                  ]),
+              ]),
+          ]),
+      ]);
+  }
+
+  void _displaySettingsDialog() {
+    // Gray Filter button while dialog is up - only apply/cancel can close.
+    settingsButton.disabled = true;
+
+    librariesUi.add(createLibrariesUi());
+    settings.element.style.display = 'block';
+
+    // Set focus to filter pattern field.
+    classNameFilter.element.focus();
+  }
+
+  void _classPatternChanged(String value) {
+    _liveUpdateFilters();
+  }
+
+  // Update the classes table (if snapshot) live.
+  void _liveUpdateFilters() {
+    final String pattern =
+        classNameFilter.value != null ? classNameFilter.value : '*';
+
+    final html.InputElement checkbox = privateClasses[0].element;
+    bool hidePrivates = checkbox.checked;
+
+    final FilteredLibraries temporaryFilters = FilteredLibraries()
```

Comment:
nit: temporaryFilters --> filter. I'm not seeing why is really temporary about this.

Reply:
Its temporary because its for a live update of the Classes Snapshot list (calls computeDisplayClasses with the tempFilter).  If user cancels settings this filter is thrown away and the original filter in the memoryController is used.